packages/app/src/cli/utilities/app-unlinked-command.ts

```diff
@@ -0,0 +1,18 @@
+import {configurationFileNames} from '../constants.js'
+import {AppInterface} from '../models/app/app.js'
+import BaseCommand from '@shopify/cli-kit/node/base-command'
+
+export interface AppUnlinkedCommandOutput {
+  app: AppInterface
+}
+
+/**
+ * This command is used to run commands that exceptionally don't require authentication or a linked app.
+ */
+export default abstract class AppUnlinkedCommand extends BaseCommand {
+  environmentsFilename(): string {
+    return configurationFileNames.appEnvironments
+  }
+
+  public abstract run(): Promise<AppUnlinkedCommandOutput>
+}
```

Comment:
Would it make sense for the Linked class to inherit from the Unlinked one? as a ways of make it more specific.
AppLinkedCommand < AppUnlinkedCommand < BaseCommand

Because `AppLinkedCommand` also conforms to the Unlinked case beacuse `AppLinkedInterface` inherits from `AppInterface`.

does that make sense? what do you think?

That way if we ever want to add something to all commands, we can still do it in `AppUnlinkedCommand`. (maybe even rename that to `AppBaseCommand`?)

Reply:
Makes sense, but maybe let's do that now then? and we can move 
```
  environmentsFilename(): string {
    return configurationFileNames.appEnvironments
  }
```
to the AppBaseCommand?
( the only thing i'll be worried in that case is commands trying to just use `AppCommand` )